clippy_lints/src/methods/mod.rs

```diff
@@ -4534,15 +4561,23 @@ impl Methods {
                 ("all", [arg]) => {
                     unused_enumerate_index::check(cx, expr, recv, arg);
                     needless_character_iteration::check(cx, expr, recv, arg, true);
-                    if let Some(("cloned", recv2, [], _, _)) = method_call(recv) {
-                        iter_overeager_cloned::check(
-                            cx,
-                            expr,
-                            recv,
-                            recv2,
-                            iter_overeager_cloned::Op::NeedlessMove(arg),
-                            false,
-                        );
+                    match method_call(recv) {
+                        Some(("cloned", recv2, [], _, _)) => {
+                            iter_overeager_cloned::check(
+                                cx,
+                                expr,
+                                recv,
+                                recv2,
+                                iter_overeager_cloned::Op::NeedlessMove(arg),
+                                false,
+                            );
+                        },
+                        Some(("map", _, [map_arg], _, map_call_span))
+                            if is_trait_method(cx, expr, sym::Iterator) && is_trait_method(cx, recv, sym::Iterator) =>
```

Comment:
It would be better to move this check into `clippy_lints/src/methods/map_all_any_identity.rs`. This module is already pretty large and having more checks on this level makes it more complex than needed.

Reply:
Ok. I was hesitant to do so because it would increase the number of arguments to the function, but I understand the rationale. Done.